packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -457,4 +474,268 @@ linter:
+
+  void _sendCommand(String method, Map<String, dynamic> params) {
+    String message = JSON.encode(<String, dynamic> {
+      'id': (++_id).toString(),
+      'method': method,
+      'params': params
+    });
+    _process.stdin.writeln(message);
+    printTrace('==> $message');
+  }
+
+  void _handleServerResponse(String line) {
+    printTrace('<== $line');
+
+    dynamic response = JSON.decode(line);
+
+    if (response is Map) {
+      if (response['event'] != null) {
+        String event = response['event'];
+        dynamic params = response['params'];
+
+        if (params is Map) {
+          if (event == 'server.status')
+            _handleStatus(response['params']);
+          else if (event == 'analysis.errors')
+            _handleAnalysisIssues(response['params']);
+          else if (event == 'server.error')
+            _handleServerError(response['params']);
+        }
+      } else if (response['error'] != null) {
+        printError('Error from the analysis server: ${response['error']['message']}');
+      }
+    }
+  }
+
+  void _handleStatus(Map<String, dynamic> statusInfo) {
+    // {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
+    if (statusInfo['analysis'] != null) {
+      bool isAnalyzing = statusInfo['analysis']['isAnalyzing'];
+      _analyzingController.add(isAnalyzing);
+    }
+  }
+
+  void _handleServerError(Map<String, dynamic> errorInfo) {
+    printError('Error from the analysis server: ${errorInfo['message']}');
+  }
+
+  void _handleAnalysisIssues(Map<String, dynamic> issueInfo) {
+    // {"event":"analysis.errors","params":{"file":"/Users/.../lib/main.dart","errors":[]}}
+    String file = issueInfo['file'];
+    List<AnalysisError> errors = issueInfo['errors'].map((Map<String, dynamic> json) => new AnalysisError(json)).toList();
+    _errorsController.add(new FileAnalysisErrors(file, errors));
+  }
+
+  Future dispose() async => _process?.kill();
+}
+
+class FileAnalysisErrors {
+  FileAnalysisErrors(this.file, this.errors);
+
+  final String file;
+  final List<AnalysisError> errors;
+}
+
+class AnalysisError implements Comparable<AnalysisError> {
+  AnalysisError(this.json);
+
+  static final Map<String, int> _severityMap = <String, int> {
+    'ERROR': 3,
+    'WARNING': 2,
+    'INFO': 1
+  };
+
+  // "severity":"INFO","type":"TODO","location":{
+  //   "file":"/Users/.../lib/test.dart","offset":362,"length":72,"startLine":15,"startColumn":4
+  // },"message":"...","hasFix":false}
+  Map<String, dynamic> json;
+
+  String get severity => json['severity'];
+  int get severityLevel => _severityMap[severity] ?? 0;
+  String get type => json['type'];
+  String get message => json['message'];
+
+  String get file => json['location']['file'];
+  int get startLine => json['location']['startLine'];
+  int get startColumn => json['location']['startColumn'];
+
+  int compareTo(AnalysisError other) {
+    // Sort in order of severity, file path, source line, and message.
```

Comment:
I've sorted this way in other tools; it's meant to sort so that the most important things to work on are first, and after that the sort is stable, and predictable for the user.

If you have have a longer list of errors - long enough to need scrolling - you don't want errors to show up below the fold, pushed down by non-critical warnings or infos. You're either less aware of errors, or end up scrolling through a list looking for an error.

I'll change to sorting by path:offset:severity and we can see how it works in the cli tool.